diff -Naur a/pkgmk.conf b/pkgmk.conf
--- a/pkgmk.conf	2018-02-14 16:17:57.000000000 +0000
+++ b/pkgmk.conf	2021-01-23 14:34:35.441690398 +0000
@@ -2,8 +2,9 @@
 # /etc/pkgmk.conf: pkgmk(8) configuration
 #
 
-export CFLAGS="-O2 -march=x86-64 -pipe"
-export CXXFLAGS="${CFLAGS}"
+export CFLAGS="-O2 -pipe -mcpu=powerpc -mtune=750 -msecure-plt" 
+#export CFLAGS="-O2 -pipe -mcpu=powerpc64 -mtune=970 -msecure-plt"
+export CXXFLAGS="$CFLAGS"
 
 # export JOBS=$(nproc)
 # export MAKEFLAGS="-j $JOBS"
@@ -39,3 +40,7 @@
 # PKGMK_COMPRESSION_MODE="gz"
 
 # End of file
+# Samples:
+# PKGMK_SOURCE_DIR="/sources"
+# PKGMK_IGNORE_FOOTPRINT="yes"
+# PKGMK_COMPRESSION_MODE="xz"
